src/Handler/TagHandler.php

```diff
@@ -88,10 +101,34 @@ static function ($tag) use ($repoPrefix) {
                 },
                 $tags
             );
+
             // Also add a un-prefixed `ez-all` in order to be able to purge all across repos
             $tags[] = 'ez-all';
 
-            $response->headers->set($this->tagsHeader, implode(' ', array_unique($tags)));
+            $tagsString = implode(' ', array_unique($tags));
+            $tagsLength = strlen($tagsString);
+            if ($this->tagsHeaderMaxLength && $tagsLength > $this->tagsHeaderMaxLength) {
+                $tagsString = substr(
+                    $tagsString,
+                    0,
+                    // Seek backwards from point of max length using negative offset
+                    strrpos($tagsString, ' ', $this->tagsHeaderMaxLength - $tagsLength)
+                );
+
+                if (
+                    $this->tagsHeaderReducedTTl &&
+                    ($responseSharedMaxAge = $response->headers->getCacheControlDirective('s-maxage')) &&
```

Comment:
Just a small remark, would you mind moving this instruction to a separate line before `if`?

Reply:
*And get rid of the micro optimization of only calling this if there is a TTL set? NEVER*

Kidding, I'll adapt it :)